pkg/gce-pd-csi-driver/controller.go

```diff
@@ -391,22 +393,34 @@ func (gceCS *GCEControllerServer) CreateVolume(ctx context.Context, req *csi.Cre
 		}
 		disk, err = createSingleZoneDisk(ctx, gceCS.CloudProvider, name, zones, params, capacityRange, capBytes, snapshotID, volumeContentSourceVolumeID, multiWriter)
 		if err != nil {
-			return nil, LoggedError("CreateVolume failed to create single zonal disk "+name+": ", err)
+			// Emit metric for expected disk type from storage class
+			if params.DiskType != "" {
```

Comment:
For the places that we don't know the disk type, should we record an error anyway with empty disktype?

Reply:
Recording metrics for unknown disk types